core/trino-spi/src/main/java/io/trino/spi/Preconditions.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+import static java.util.Objects.requireNonNull;
+
+public class Preconditions
```

Comment:
this should be `final`, per https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#additional-ide-configuration

However, this should also be package-private, we don't want to make this part of SPI's contract.
Since this is used shared between `io.trino.spi.ptf` and `io.trino.spi` it cannot be package-private.

Please add this as a package-private utility class in `io.trino.spi.ptf` package, and use it only there.
Revert `io.trino.spi.procedure` changes.